plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -325,9 +331,21 @@ && predicateMatches(parquetPredicate, block, dataSource, descriptorsByPath, parq
         return new ReaderPageSource(parquetPageSource, readerProjections);
     }
 
-    public static Optional<org.apache.parquet.schema.Type> getParquetType(GroupType groupType, boolean useParquetColumnNames, HiveColumnHandle column)
+    private static ColumnIO getColumnIo(HiveColumnHandle column, MessageColumnIO messageColumn, MessageType fileSchema, ParquetColumnMapping columnMapping)
     {
-        if (useParquetColumnNames) {
+        if (columnMapping == FIELD_ID) {
+            return lookupColumnById(messageColumn, column.getBaseHiveColumnIndex());
+        }
+        String columnName = columnMapping == NAME ? column.getBaseColumnName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
+        return lookupColumnByName(messageColumn, columnName);
+    }
+
+    private static Optional<org.apache.parquet.schema.Type> getParquetType(GroupType groupType, ParquetColumnMapping columnMapping, HiveColumnHandle column)
+    {
+        if (columnMapping == FIELD_ID) {
+            return Optional.ofNullable(getParquetTypeById(column.getBaseHiveColumnIndex(), groupType));
+        }
+        if (columnMapping == NAME) {
             return Optional.ofNullable(getParquetTypeByName(column.getBaseColumnName(), groupType));
         }
         if (column.getBaseHiveColumnIndex() < groupType.getFieldCount()) {
```

Comment:
Is this the case for `ParquetColumnMapping.COLUMN_INDEX` ?

Would it make sense to use a `switch` statement here ?